### PR TITLE
Stboot wait for harddisks

### PR DESCRIPTION
**Pull request by PatrickRudolph**
Wait for non loopback devices to appear.
On real hardware this takes about 3 seconds after stboot has started.
Improve error message while on it.

### CODE REVIEW

File: cmds/boot/stboot/data.go [hunk 1]
```diff
@@ -36,22 +37,33 @@ func findDataPartition() error {
...
 	}
 
-	devices, err := getBlockDevs()
-	if err != nil {
-		return fmt.Errorf("block devices: %v", err)
-	}
-	if len(devices) == 0 {
-		return fmt.Errorf("no non-loopback block devices found")
+	var devices []string
+	for {
```
Comment (rminnich) [inline cmds/boot/stboot/data.go:41]: for len(devices) == 0 {
maybe. Then you can skip the break?

Comment (rminnich) [reply cmds/boot/stboot/data.go:41]: ah ... for len(devices) == 0 || timeout > 0

File: cmds/boot/stboot/data.go [hunk 2]
```diff
@@ -36,22 +37,33 @@ func findDataPartition() error {
...
+			return fmt.Errorf("getting block devices failed with: %v", err)
+		}
+		if len(devices) == 0 {
+			if timeout == 0 {
+				return fmt.Errorf("no non-loopback block devices found")
+			}
+			debug("waiting for block devices to appear...\n")
+
+			timeout--
+			time.Sleep(time.Second)
```
Comment (rminnich) [inline cmds/boot/stboot/data.go:53]: a second? ow. This is usually a firmware scenario, one process, multiple cores, burn the core. 10 ms? 1?

**Comment by hugelgupf** [comment]
`pkg/storage` has moved to `pkg/mount/block`

**Comment by jensdrenhaus** [comment]
It's all work in progress and only merged into the stboot branch. I need to make adaptions anyway regarding other packages too.
Thanks for your comments. We will check before issuing a PR into master.

**Pull request merged**